src/db/sqlite_internals/sqlite_proposal.rs

```diff
@@ -0,0 +1,120 @@
+use rusqlite::Row;
+
+use crate::{
+    db::{
+        shared::{DbError, ProposalAdapter},
+        sqlite::SQLite,
+    },
+    models::Proposal,
+};
+
+impl ProposalAdapter for SQLite {
+    fn get_proposal(&self, title: &str, owner: &str) -> Result<Proposal, DbError> {
+        let mut query = self
+            .connection()
+            .unwrap()
+            .prepare(
+                "SELECT 
+                title,
+                description, 
+                created, 
+                updated, 
+                comments, 
+                owner, 
+                upvotes 
+                FROM proposals WHERE title = (?1) AND owner = (?2)",
+            )
+            .unwrap();
+        let mut res = query.query_map([title, owner], db_row_to_proposal).unwrap();
+
+        res.next().unwrap().map_err(|op| DbError {
+            message: format!("{} -> {}", String::from("failed to find proposal"), op),
+        })
+    }
+
+    fn insert_proposal(&self, p: Proposal) -> Result<Proposal, DbError> {
+        let query = self.connection().unwrap().execute(
+            "INSERT INTO proposals (
+                title, 
+                description, 
+                created, 
+                updated, 
+                comments, 
+                owner, 
+                upvotes
+            ) values (?1, ?2, ?3, ?4, ?5, ?6, ?7)",
+            [
+                &p.title,
+                &p.description,
+                &p.created.to_string(),
+                &p.updated.to_string(),
+                &p.comments.join(","),
+                &p.owner,
+                &p.upvotes.join(","),
+            ],
+        );
+
+        match query {
+            Ok(_) => Ok(p),
+            Err(err) => {
+                eprintln!("{}", err);
+
+                Err(DbError {
+                    message: format!("failed to insert {:?}", p),
+                })
+            }
+        }
+    }
```

Comment:
And this could be changed into something like this:
```rust
    fn insert_proposal(&self, p: Proposal) -> Result<Proposal, Box<dyn Error>> {
        let _ = self.connection()?.execute(
            "INSERT INTO proposals (
                id,
                title, 
                description, 
                created, 
                updated, 
                comments, 
                owner, 
                upvotes
            ) values (?1, ?2, ?3, ?4, ?5, ?6, ?7)",
            [
                &p.id,
                &p.title,
                &p.description,
                &p.created.to_string(),
                &p.updated.to_string(),
                &p.comments.join(","),
                &p.owner,
                &p.upvotes.join(","),
            ],
        )?;
        Ok(p)
    }
```

Reply:
I think an `id` on `Proposal` might be a nice utilty. Maybe not as an incremented integer but perhaps a UUID stored as a string. There is a crate named `uuid` that looks ok.